public/pages/AnomalyCharts/components/FeatureChart/FeatureChart.tsx

```diff
@@ -240,15 +261,35 @@ export const FeatureChart = (props: FeatureChartProps) => {
             showGridLines
           />
           <Axis id="bottom" position="bottom" tickFormat={timeFormatter} />
-          <LineSeries
-            id="featureData"
-            name={props.featureDataSeriesName}
-            xScaleType={ScaleType.Time}
-            yScaleType={ScaleType.Linear}
-            xAccessor={CHART_FIELDS.PLOT_TIME}
-            yAccessors={[CHART_FIELDS.DATA]}
-            data={featureData}
-          />
+          {
+            // Add each set of feature data as a separate time series
+          }
+          {featureData.map(
+            (featureTimeSeries: FeatureAggregationData[], index) => {
+              const seriesKey = props.isHCDetector
```

Comment:
Might be misunderstanding typescript 'types' but since `isHCDetector` is an optional, means `props.isHCDetector` is a 'falsy' value and lead to the case after `:`. Is this on purpose or could it lead to any issues.

Reply:
It's on purpose - if `isHCDetector` exists and is `true`, then the true condition runs; else the false condition runs.